### PR TITLE
[l10n] Add Macedonian (mk) locale

### PR DESCRIPTION
**Pull request by brsnik**
- [x] I have followed (at least) the [PR section of the contributing guide](https://github.com/mui/mui-x/blob/HEAD/CONTRIBUTING.md#sending-a-pull-request).

### CODE REVIEW

**Review by brsnik — COMMENTED** [review]
Removed the default setting

**Review by brsnik — COMMENTED** [review]
Checklist

> * [x]  Verify if the PR title respects the release format. Here are two examples (depending if you update or add a locale file)
>   > [l10n] Improve Swedish (sv-SE) locale
>   > [l10n] Add Danish (da-DK) locale
> * [x]  Update the documentation of supported locales by running `yarn l10n`
> * [x]  Verify that you have added an export line in `src/locales/index.ts` for the new locale.
> * [x]  Run `yarn docs:api` which should add your new translation to the list of exported interfaces.
> * [x]  Clean files with `yarn prettier`.

**Comment by flaviendelangle** [comment]
@LukasTy how do you think we should handle that one?

**Comment by LukasTy** [comment]
> how do you think we should handle that one?

Sorry, completely forgot this PR during the retreat. 🙈 
IMHO, it's another perfect candidate to introduce without the country suffix alongside the `Basque` (eu) locale. 👌

**Comment by LukasTy** [comment]
Hello @brsnik.
We are changing the direction of the localization a bit.
I wanted to adjust your PR accordingly, but it is created from your default fork branch, which is protected and I can't push to it. 🙈 
Could you merge the latest `mui-x/next` branch into your branch, apply the following diff and re-run `yarn l10n && docs:api`? 🤔 🙏 

```diff
diff --git a/packages/x-date-pickers/src/locales/index.ts b/packages/x-date-pickers/src/locales/index.ts
index ac79492bb..bc85adf53 100644
--- a/packages/x-date-pickers/src/locales/index.ts
+++ b/packages/x-date-pickers/src/locales/index.ts
@@ -16,6 +16,7 @@ export * from './itIT';
 export * from './jaJP';
 export * from './koKR';
 export * from './kzKZ';
+export * from './mk';
 export * from './nbNO';
 export * from './nlNL';
 export * from './plPL';
@@ -30,5 +31,4 @@ export * from './urPK';
 export * from './viVN';
 export * from './zhCN';
 export * from './zhHK';
-export * from './mkMK';
 export * from './utils/pickersLocaleTextApi';
diff --git a/packages/x-date-pickers/src/locales/mkMK.ts b/packages/x-date-pickers/src/locales/mk.ts
similarity index 96%
rename from packages/x-date-pickers/src/locales/mkMK.ts
rename to packages/x-date-pickers/src/locales/mk.ts
index a6fee0c2b..ec45ce7f5 100644
--- a/packages/x-date-pickers/src/locales/mkMK.ts
+++ b/packages/x-date-pickers/src/locales/mk.ts
@@ -3,7 +3,7 @@ import { getPickersLocalization } from './utils/getPickersLocalization';
 
 // This object is not Partial<PickersLocaleText> because it is the default values
 
-const mkMKPickers: PickersLocaleText<any> = {
+const mkPickers: PickersLocaleText<any> = {
   // Calendar navigation
   previousMonth: 'Предходен месец',
   nextMonth: 'Следен месец',
@@ -76,4 +76,4 @@ const mkMKPickers: PickersLocaleText<any> = {
   fieldMeridiemPlaceholder: () => 'aa',
 };
 
-export const mkMK = getPickersLocalization(mkMKPickers);
+export const mk = getPickersLocalization(mkPickers);
diff --git a/scripts/localeNames.js b/scripts/localeNames.js
index 24ff47a66..08757be67 100644
--- a/scripts/localeNames.js
+++ b/scripts/localeNames.js
@@ -33,7 +33,7 @@ module.exports = {
   'kh-KH': 'Khmer',
   'kz-KZ': 'Kazakh',
   'ko-KR': 'Korean',
-  'mk-MK': 'Macedonian',
+  mk: 'Macedonian',
   'nb-NO': 'Norwegian (Bokmål)',
   'fa-IR': 'Persian',
   'pl-PL': 'Polish',

```

**Comment by brsnik** [comment]
@LukasTy Done. Please let me know if you need anything else.

**Comment by LukasTy** [comment]
> Done. Please let me know if you need anything else.

Thank you! 🙏 
Could you please apply the rest of the mentioned diff and after that run `yarn l10n`?
You can apply the diff by copying it into a file (i.e. named `diff.patch`) in the root of your project and then running `git apply <the name of the file you created>`.
Or by simply applying the remaining diff:
- move the export in the alphabetical order (nitpick)
- rename the `Macedonian` locale

File: packages/grid/x-data-grid-generator/src/services/static-data.ts [hunk 1]
```diff
@@ -229,7 +229,7 @@ export const COUNTRY_ISO_OPTIONS: CountryIsoOption[] = [
   { value: 'MF', code: 'MF', label: 'Saint Martin (French part)', phone: '590' },
   { value: 'MG', code: 'MG', label: 'Madagascar', phone: '261' },
   { value: 'MH', code: 'MH', label: 'Marshall Islands', phone: '692' },
-  { value: 'MK', code: 'MK', label: 'Macedonia, the Former Yugoslav Republic of', phone: '389' },
+  { value: 'MK', code: 'MK', label: 'North Macedonia', phone: '389' },
```
Comment (LukasTy) [inline packages/grid/x-data-grid-generator/src/services/static-data.ts:232]: Nice additional fix. 👏

File: scripts/localeNames.js [hunk 2]
```diff
@@ -33,7 +33,7 @@ module.exports = {
   'kh-KH': 'Khmer',
   'kz-KZ': 'Kazakh',
   'ko-KR': 'Korean',
-  'mk-MK': 'Macedonian',
+  'mk': 'Macedonian',
```
Comment (LukasTy) [inline scripts/localeNames.js:36]:
```suggestion
  mk: 'Macedonian',
```

**Review by LukasTy — APPROVED** [review]
@brsnik Thank you so much for your contribution and patience with providing the necessary adjustments! 🙏 💯 🎉

**Comment by brsnik** [comment]
> @brsnik Thank you so much for your contribution and patience with providing the necessary adjustments! 🙏 💯 🎉

@LukasTy Anytime! :)

File: packages/x-date-pickers/src/locales/mk.ts [hunk 3]
```diff
@@ -0,0 +1,79 @@
+import { PickersLocaleText } from './utils/pickersLocaleTextApi';
+import { getPickersLocalization } from './utils/getPickersLocalization';
+
+// This object is not Partial<PickersLocaleText> because it is the default values
+
+const mkPickers: PickersLocaleText<any> = {
+  // Calendar navigation
+  previousMonth: 'Предходен месец',
```
Comment (mnajdova) [inline packages/x-date-pickers/src/locales/mk.ts:8]:
```suggestion
  previousMonth: 'Претходен месец',
```

I stumbled on this PR and couldn't help but review :D @brsnik would you like to create a follow PR to fix this? :)

Comment (brsnik) [reply packages/x-date-pickers/src/locales/mk.ts:8]: @mnajdova Will do.

**Pull request merged**